File: models/Menu.java
package models;
public class Menu {
    private int id;
    private String name;
    private String iamge;
    private Burger burger;
    private Complement boisson;
    private Complement frites;

    public Menu(int id, String name, String iamge, Burger burger, Complement boisson, Complement frites) {
        this.id = id;
        this.name = name;
        this.iamge = iamge;
        this.burger = burger;
        this.boisson = boisson;
        this.frites = frites;
    }

    public double getPrix() {
        return burger.getPrix() + boisson.getPrix() + frites.getPrix();
    }

    public String getNom() { return nom; }
}

File: Models/Menu.cs
using System.ComponentModel.DataAnnotations;

namespace BrasilBurger.Models
{
    public class Menu
    {
        public int Id { get; set; }

        [Required]
        public required string Nom { get; set; }

        public required string Image { get; set; }

        [Required]
        public int BurgerId { get; set; }

        public Burger? Burger { get; set; }

        // For simplicity, assume menu includes boisson and frites as complements
        public ICollection<MenuComplement> MenuComplements { get; set; } = new List<MenuComplement>();

        // Prix calculated as sum
        public decimal Prix => Burger?.Prix + MenuComplements.Sum(mc => mc.Complement?.Prix ?? 0) ?? 0;

        public bool Archive { get; set; } = false;
    }

    public class MenuComplement
    {
        public int Id { get; set; }

        public int MenuId { get; set; }

        public Menu? Menu { get; set; }

        public int ComplementId { get; set; }

        public Complement? Complement { get; set; }
    }
}